### PR TITLE
Fix IPv6 format

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Review by shargon — DISMISSED** [review]

File: src/NameService/NameService.cs [hunk 1]
```diff
@@ -504,9 +505,10 @@ private static bool CheckIPv6(string ipv6)
                 {
                     if (fragment.Length > 4) return false;
                     int index = hasEmpty ? i + 8 - length : i;
-                    numbers[index] = (ushort)StdLib.Atoi(fragment, 16);
+                    numbers[index] = (ushort)(short)StdLib.Atoi(fragment, 16);
```
Comment (shargon) [inline src/NameService/NameService.cs:508]: I think that if is negative we should throw an exception

Comment (erikzhang) [reply src/NameService/NameService.cs:508]: `"ffff"` will be parsed to `-1`. So we need to convert it.

**Review by shargon — APPROVED** [review]

**Comment by erikzhang** [comment]
@cloud8little

**Review by cloud8little — APPROVED** [review]

File: src/NameService/NameService.cs [hunk 2]
```diff
@@ -474,12 +474,13 @@ private static bool CheckIPv6(string ipv6)
...
             if (length < 3 || length > 8) return false;
             ushort[] numbers = new ushort[8];
-            bool hasEmpty = false;
+            bool hasEmpty = false, isCompressed = false;
             for (int i = 0; i < length; i++)
             {
                 string fragment = fragments[i];
                 if (fragment.Length == 0)
                 {
+                    isCompressed = true;
```
Comment (roman-khimov) [inline src/NameService/NameService.cs:483]: IIUC this will allow for `aabb:` address which in fact is invalid (while `aabb::` is).

Comment (erikzhang) [reply src/NameService/NameService.cs:483]: Please check #7

**Pull request merged**